### ISSUE TITLE
Make saga serializer configurable

### ISSUE THREAD

**Issue by mathiasnohall** [rebus-org/Rebus.SqlServer#59]
if I use the configurationbuilder like so:
```C#
.Serialization(s => s.UseNewtonsoftJson(JsonInteroperabilityMode.PureJson))
```
the objectserializer still uses json with full type information. i.eg. when persisting saga data to the sql database in SqlServerSagaStorage.cs

the NewtonsoftJsonConfigurationExtensions should register an instance of ObjectSerializer wich honors the purejson interopability

**Comment by mookid8000** [comment]
The SQL saga persister will serialize data using Rebus' built-in `ObjectSerializer`, which is full of type information. 

While this is convenient to be able to reliably roundtrip objects involving inheritance hierarchies and stuff, it's not really well-suited for long-lived saga data, as it makes refactoring the saga code harder.

Therefore, it would be nice if the saga data serializer could be configured.

The API should probably be changed so that something like the following could be done:
```csharp
Configure.With(...)
	.(...)
	.Sagas(s => {
		s.StoreInSqlServer(...)
			.SerializeSagaDataWith(new MyCustomSagaDataSerializer());
	})
	.Start();
```
which requires that `StoreInSqlServer` returns a `SagaConfigurationCustomizer` that can be used for further customizations.

**Comment by mookid8000** [comment]
Fixed by #77
